src/modules/nav/nav.service.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { Router, NavigationStart, NavigationEnd } from '@angular/router';
+
+import { Observable } from 'rxjs';
+
+import { StacheNavService } from './nav.service';
+
+class MockSkyAppConfig {
+  public runtime: any = {
+    routes: [
+      {
+        routePath: ''
+      },
+      {
+        routePath: 'parent'
+      },
+      {
+        routePath: 'parent/child'
+      },
+      {
+        routePath: 'parent/child/grandchild'
+      },
+      {
+        routePath: 'parent/child/grandchild/grand-grandchild'
+      },
+      {
+        routePath: 'other-route'
+      },
+      {
+        routePath: 'other-parent'
+      },
+      {
+        routePath: 'other-parent/other-child'
+      },
+      {
+        routePath: 'other-parent/other-child/other-grandchild'
+      }
+    ]
+  };
+  public skyux: any = {};
+}
+
+class MockRouter {
+  public url = '/parent/child/grandchild';
+  public events = Observable.of(new NavigationStart(0, ''));
+}
+
+describe('StacheNavService', () => {
+  let navService: StacheNavService;
+  let skyAppConfig: MockSkyAppConfig;
+  let router: MockRouter;
+
+  beforeEach(() => {
+    skyAppConfig = new MockSkyAppConfig();
+    router = new MockRouter();
+    navService = new StacheNavService(skyAppConfig, router as Router);
+  });
+
+  it('should only assemble the active routes once', () => {
+    let activeRoutes = navService.getActiveRoutes();
+    expect(activeRoutes[0].children.length).toBe(1);
+
+    skyAppConfig.runtime.routes = [];
+    activeRoutes = navService.getActiveRoutes();
+    expect(activeRoutes[0].children.length).toBe(1);
+  });
+
+  it('should only unset the active routes on NavigationStart', () => {
+    router.events = Observable.of(new NavigationEnd(0, '', ''));
```

Comment:
Ha! Yeah, this was fun. It's the equivalent to ng1's router hooks. I wanted to clear the active routes when the user navigates. This test was the only way to reach a specific branch in the coverage.